lib/screens/explore_screens/regions_screens/region_event_details/interactor/viewmodels/region_event_details_bloc.dart

```diff
@@ -18,4 +25,32 @@ class RegionEventDetailsBloc extends Bloc<RegionEventDetailsEvent, RegionEventDe
     on<OnDeleteEventTapped>((_, emit) => emit(state.copyWith()));
     on<ClearRegionEventPageCommand>((_, emit) => emit(state.copyWith()));
   }
+
+  Future<void> _onJoinRegionEventButtonPressed(
+      OnJoinRegionEventButtonPressed event, Emitter<RegionEventDetailsState> emit) async {
+    emit(state.copyWith(pageState: PageState.loading));
+    final result = await JoinRegionEventUseCase().run(JoinRegionEventUseCase.input(
+      eventId: state.event.id,
+      joiningUser: settingsStorage.accountName,
+    ));
+    if (result.isError) {
+      emit(state.copyWith(pageCommand: ShowErrorMessage('')));
```

Comment:
do we want to show an empty error?

Reply:
For this part since it is Firebase call I do not know if we should show a more detailed error.
I'm handling it with a snackbar that says _Operation failed try again_. What do you think should change the snack for the full error screen?

Reply:
snackbar is fine.